### PR TITLE
Allow resetPassword by email only if email verification was done

### PR DESCRIPTION
**Pull request by loay**
connect to https://github.com/strongloop/loopback/issues/382

### CODE REVIEW

**Comment by loay** [comment]
@raymondfeng @bajtos PTAL. Thanks

File: common/models/user.js [hunk 1]
```diff
@@ -560,19 +560,26 @@ module.exports = function(User) {
...
         return cb(err);
       }
-      // create a short lived access token for temp login to change password
-      // TODO(ritch) - eventually this should only allow password change
-      user.accessTokens.create({ ttl: ttl }, function(err, accessToken) {
-        if (err) {
-          return cb(err);
-        }
-        cb();
+      if (user && user.emailVerified) {
```
Comment (bajtos) [inline common/models/user.js]: `user` is always truthy, see L557 above. no need to check it here.

File: common/models/user.js [hunk 2]
```diff
@@ -560,19 +560,26 @@ module.exports = function(User) {
...
         UserModel.emit('resetPasswordRequest', {
           email: options.email,
-          accessToken: accessToken,
```
Comment (bajtos) [inline common/models/user.js]: How are you going to construct an authenticated URL when there is no accessToken available?

**Comment by bajtos** [comment]
@loay let's keep this pull request focused on the first task from #382, which is

> Only allow resetPassword by email if email verification is required and done.

The change proposed here goes in the opposite direction. Here is what we need:

1) if email is not verified, then return an error
2) if email was verified, then continue with the same behaviour as before

**Pull request closed without merging**